fe/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -1244,6 +1213,46 @@ public void rewriteExprs(ExprRewriter rewriter) throws AnalysisException {
                 orderByElem.setExpr(rewriter.rewrite(orderByElem.getExpr(), analyzer));
             }
         }
+        if (subqueryInCase) {
+            for (SelectListItem item : selectList.getItems()) {
+                if (!(item.getExpr() instanceof CaseExpr)) {
+                    continue;
+                }
+                if (!item.getExpr().contains(Predicates.instanceOf(Subquery.class))) {
+                    continue;
+                }
+                item.setExpr(rewriteSubquery(item.getExpr()));
+            }
+        }
+    }
+
+
+    private Expr rewriteSubquery(Expr expr) throws AnalysisException {
+        if (expr instanceof Subquery) {
+            if (!(((Subquery) expr).getStatement() instanceof SelectStmt) ) {
+                throw new AnalysisException("only support select subquery in case statement.");
+            }
+            SelectStmt subquery = (SelectStmt)((Subquery) expr).getStatement();
+            if (subquery.getSelectList().getItems().size() != 1) {
```

Comment:
How do you make sure the subquery is like `select sum(k1) from tbl`, not `select k1 from tbl`?

Reply:
this expr is validate by the predicate of case when clause